### PR TITLE
Rework the API, and the documentation, and everything really.

### PR DESCRIPTION
**Pull request by Drup**
This is built on top of #94

This is a big cleanup/rework of the various APIs to make them easier to use. This also deprecates a bunch of stuff.

The manual part is not finished yet, but feel free to comment, and tell me what is good and what you would like to see there.

This is breaking, obviously ...

### CODE REVIEW

**Comment by Drup** [comment]
The new documentation is readable [here](http://ocsigen.org/tyxml/dev/manual/intro), various modules received improvements too. 

I would gladly appreciate some native english spaker to check my frenglish. Thanks! :]

File: doc/manual-wiki/intro.wiki [hunk 1]
```diff
@@ -1,23 +1,134 @@
...
+==@@id="usage"@@ Using TyXML ==
+
+=== Standalone Use ===
+
+To use TyXML in standalone manner, simply link the ##tyxml## ocamlfind package
+and use the <<a_api|module Html5>> and <<a_api|module Svg>> modules.
+
+=== Use with another library ===
+
+TyXML combinators can be used in conjunction to another library, please consult the relevent document. For example, for
```
Comment (rgrinberg) [inline doc/manual-wiki/intro.wiki]: typo: relevent

Comment (rgrinberg) [inline doc/manual-wiki/intro.wiki]: "in conjunction with other libraries"

File: doc/manual-wiki/intro.wiki [hunk 2]
```diff
@@ -1,23 +1,134 @@
...
+<<a_api project="js_of_ocaml" text="Js_of_ocaml"|module Tyxml_js>>.
+
+==@@id="creation"@@ Creating documents with TyXML ==
+
+This section assumes you have at your disposal an {{{Html5}}} or {{{Svg}}} module, as instructed in <<a_manual fragment="usage"|the previous section>>.
+For standalone use, examples are available in the [[https://github.com/ocsigen/tyxml/tree/master/examples|examples]] directory.
+
+The documentation for TyXML combinators is provided in <<a_api|modtype Html5_sigs.T>> and <<a_api|module Svg_sigs.T>> and is common to all instances of {{{Html5}}} and {{{Svg}}}.
+
+A first thing to understand about TyXML is that for most intent and purposes, it is exactly like Html. As such, the [[https://developer.mozilla.org/en-US/docs/Web/HTML/Element|Html reference]] is still very useful. For each Html elements or attributes, there is a combinator implemeting it. The main differences are that you can use OCaml to manipulate the elements and that invalid Html produces a type error.
```
Comment (rgrinberg) [inline doc/manual-wiki/intro.wiki]: The first

Comment (rgrinberg) [inline doc/manual-wiki/intro.wiki]: intents

Comment (rgrinberg) [inline doc/manual-wiki/intro.wiki]: implementing

**Comment by Drup** [comment]
@rgrinberg All fixed, thanks!

**Pull request merged**